modules/cli/src/main/scala/scala/cli/ScalaCli.scala

```diff
@@ -272,7 +272,18 @@ object ScalaCli {
             val newArgs = powerArgs ++ finalScalaRunnerArgs ++ args0
             LauncherCli.runAndExit(ver, launcherOpts, newArgs)
           case _ if
-                javaMajorVersion < Constants.minimumBloopJavaVersion
+                javaMajorVersion < Constants.minimumLauncherJavaVersion
+                && sys.props.get("scala-cli.kind").exists(_.startsWith("jvm")) =>
+            System.err.println(
+              s"[${Console.RED}error${Console.RESET}] Java $javaMajorVersion is not supported with this Scala CLI (JVM) launcher."
+            )
+            System.err.println(
+              s"[${Console.RED}error${Console.RESET}] Please upgrade to at least Java ${Constants.minimumLauncherJavaVersion} or use a native Scala CLI launcher instead."
```

Comment:
Note that JVM launchers of Scala CLI will now hard enforce Java 11+.
This is perhaps unsurprising, since we build Scala CLI with Java 17, and the fallback has been provided for convenience' sake.
The fallback code from `coursier` now requires Java 11+, however, and so will we.